service/forward_service.cc

```diff
@@ -0,0 +1,176 @@
+            }
+            default:
+                throw std::runtime_error("not implemented reduction type");
+        }
+    }();
+
+    // create pager object
+    auto now = gc_clock::now();
+    auto page_size = req.page_size;
+    auto timeout = req.timeout;
+    auto cql_serialization_format = cql_serialization_format::latest_version;
+    std::vector<size_t> empty_group_by_indices;
+    std::vector<cql3::raw_value> empty_values;
+    std::optional<std::vector<sstring_view>> empty_names;
+    auto query_state = make_lw_shared<service::query_state>(client_state::for_internal_calls(), nullptr,
+                                                            empty_service_permit());
+    auto query_options = make_lw_shared<cql3::query_options>(cql3::default_cql_config, req.cl, empty_names,
+                                                                empty_values, req.skip_metadata,
+                                                                cql3::query_options::specific_options::DEFAULT,
+                                                                cql_serialization_format::latest());
+    auto p = service::pager::query_pagers::pager(schema, selection, *query_state, *query_options, command,
+                                                    std::move(req.pr), nullptr);
+    auto builder = cql3::selection::result_set_builder(*selection, now, cql_serialization_format::latest(), std::move(empty_group_by_indices));
+
+    // execute query
+    while (!p->is_exhausted()) {
+        flogger.debug("forward_request: fetching page");
+        co_await p->fetch_page(builder, page_size, now, timeout);
+    }
+    co_return co_await builder.with_thread_if_needed([this, &p, &builder] {
+        flogger.debug("forward_request: building result");
+        auto rs = builder.build();
+        auto& rows = rs->rows();
+        if (rows.size() != 1) {
+            flogger.error("forward_request: result row count != 1");
+            throw std::runtime_error("aggregation result row count != 1");
+        }
+        if (rows[0].size() != 1) {
+            flogger.error("forward_request: result column count != 1");
+            throw std::runtime_error("aggregation result column count != 1");
+        }
+        auto& result_bytes = rows[0][0];
+        query::forward_result res = { .query_response = result_bytes };
+        return res;
+    });
+}
+
+void forward_service::init_messaging_service() {
+    auto& ms = _messaging;
+    ms.register_forward_request([this](query::forward_request req) -> future<query::forward_result> {
+        return execute(req);
+    });
+}
+
+future<> forward_service::uninit_messaging_service() {
+    return _messaging.unregister_forward_request();
+}
+
+future<query::forward_result> forward_service::dispatch(query::forward_request req) {
+    schema_ptr schema = local_schema_registry().get(req.cmd.schema_version);
+    keyspace& ks = _db.local().find_keyspace(schema->ks_name());
+
+    (void)_shared_token_metadata;
```

Comment:
Why this?

Reply:
Sneaked in there by accident, removed in v3